src/main/scala/viper/server/vsi/MessageStreamingTask.scala

```
@@ -40,8 +41,10 @@ abstract class MessageStreamingTask[T]() extends Callable[T] with Post {
   }
 
   private var q_actor: ActorRef = _
+  private var hasEnded: Boolean = false
 
   final def setQueueActor(actor: ActorRef): Unit = {
```

Also here: better to throw an exception.